lexos/views/content_analysis_view.py

```diff
@@ -0,0 +1,209 @@
+import json
+
+from flask import request, session, render_template, Blueprint
+
+from lexos.managers.utility import load_file_manager
+from lexos.models.content_analysis_model import ContentAnalysisModel
+from lexos.views.base_view import detect_active_docs
+
+# this is a flask blue print
+# it helps us to manage groups of views
+# see here for more detail:
+# http://exploreflask.com/en/latest/blueprints.html
+# http://flask.pocoo.org/docs/0.12/blueprints/
+content_analysis_blueprint = Blueprint('content_analysis', __name__)
+analysis = None
+
+
+# Tells Flask to load this function when someone is at '/contentanalysis'
+@content_analysis_blueprint.route("/contentanalysis", methods=["GET", "POST"])
+def content_analysis():
+    """Handles the functionality on the contentanalysis page.
+
+    :return: a response object (often a render_template call) to flask and
+    eventually to the browser.
+    """
+    global analysis
+    if analysis is None or 'content_analysis' not in session:
+        analysis = ContentAnalysisModel()
+        session['content_analysis'] = True
+    elif len(analysis.corpus) == 0:
+        file_manager = load_file_manager()
+        active_files = file_manager.get_active_files()
+        for file in active_files:
+            analysis.add_corpus(file_name=file.name,
+                                label=file.label,
+                                content=file.load_contents())
+    if request.method == 'GET':
+        dictionary_labels = []
+        active_dictionaries = []
+        session['toggle_all'] = False
+        if len(analysis.dictionaries):
+            session['toggle_all'] = True
+            for dictionary in analysis.dictionaries:
+                dictionary_labels.append(dictionary.label)
+                active_dictionaries.append(dictionary.active)
+                if not dictionary.active:
+                    session['toggle_all'] = False
+        return render_template('contentanalysis.html',
+                               dictionary_labels=dictionary_labels,
+                               active_dictionaries=active_dictionaries,
+                               toggle_all=session['toggle_all'])
+    else:
+        formula = request.json['calc_input']
+        if len(formula) == 0:
+            session['formula'] = "0"
+        else:
+            formula = formula.replace("√", "sqrt").replace("^", "**")
+            session['formula'] = formula
+            error_msg = "Formula errors:<br>"
```

Comment:
if you want to send an error to the frontend, raise an "LexosException", see this function: https://github.com/WheatonCS/Lexos/blob/master/lexos/application.py#L111